modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -84,16 +85,37 @@ export class Transaction extends BaseTransaction {
    * Get the recipient account and the amount
    * transferred on this transaction
    *
-   * @returns {[string, string]} - First element is the recipient, second element is the amount
+   * @returns {{recipient: string; amount: string; tokenName?: string}} consisting
+   *  the recipient, the transfer amount, and the token name for token transfer
    */
-  private getTransferData(): [string, string] {
+  private getTransferData(): { recipient: string; amount: string; tokenName?: string } {
     let transferData;
-    this._txBody.cryptoTransfer!.transfers!.accountAmounts!.forEach((transfer) => {
-      const amount = Long.fromValue(transfer.amount!);
-      if (amount.isPositive()) {
-        transferData = [stringifyAccountId(transfer.accountID!), amount.toString()];
-      }
-    });
+    const tokenTransactions: proto.ITokenTransferList[] = this._txBody.cryptoTransfer?.tokenTransfers || [];
+    if (tokenTransactions.length > 0) {
```

Comment:
The if/else code blocks here are redundant.
Try simplifying it something like this:
```
    let transferData;
    const tokenTransfers: proto.ITokenTransferList[] = this._txBody.cryptoTransfer?.tokenTransfers || [];

    const transfers: proto.IAccountAmount[] =
      tokenTransfers[0]?.transfers || this._txBody.cryptoTransfer?.transfers?.accountAmounts || [];

    transfers.forEach((transfer) => {
      const amount = Long.fromValue(transfer.amount!);
      if (amount.isPositive()) {
        transferData = {
          recipient: stringifyAccountId(transfer.accountID!),
          amount: amount.toString(),
          ...(tokenTransfers.length && {
            tokenName: getHbarTokenFromId(stringifyTokenId(tokenTransfers[0].token!), this._coinConfig.network)?.name,
          }),
        };
      }
    });

    return transferData;
```